code/helpers/vector.dm

```diff
@@ -77,6 +82,14 @@ Inputs:
 		coord_x += inc_x
 		coord_y += inc_y
 
+		// The spinning x strikes the y in the z!
+		if(spin_counter != -1)
+			spin_counter++
+			if(spin_counter >= 8) spin_counter = 0
+			var/matrix/M = matrix()
+			M.Turn(90 * round(spin_counter/2))
```

Comment:
I believe we can actually use animate() instead and just apply it once on loop to achieve the same effect. Animate() has the bonus of being a client side effect, so doesn't add any load on the server.